### ISSUE TITLE
SIM108 not raised for instance variables

### ISSUE THREAD

**Issue by cspoelstra** [astral-sh/ruff#16542]
### Summary

SIM108 is not being raised for instance variables.

```
class A:
    def func(self):
        if cond:
            self.x = a
        else:
            self.x = b
```

Expected behavior:
SIM108 raised to suggest
```
self.x = a if cond else b
```
Actual behavior:
SIM108 not raised.

Removing `self.` causes the rule to be raised.

**Comment by MichaReiser** [comment]
The challenge with allowing more expressions is that we then get pushback that the resulting expression is now too complicated and isn't a simplification. That's why, for now, I'd prefer to keep the rule aligned with the upstream rule.
